cmd/validator/core/validator.go

```diff
@@ -119,7 +129,101 @@ func Start(work <-chan string, processorID ProcessorID, threads int, config Para
 }
 
 // CallProcessor invokes the processor with a retry mechanism.
-func CallProcessor(processor Processor, addrInput string, config Params, results chan<- Result) {
+func CallProcessor(processor Processor, input string, config Params, results chan<- Result) {
+	split := strings.Split(input, ",")
+	if len(split) == 1 {
+		callProcessorAddress(processor, split[0], config, results)
+	} else {
+		if split[0] == "address" && len(split) == 2 {
+			callProcessorAddress(processor, split[1], config, results)
+		} else if split[0] == "box" && len(split) == 3 {
+			callProcessorBox(processor, split[1], split[2], config, results)
+		} else {
+			err := fmt.Errorf("invalid resource")
+			results <- resultResourceError(err, input)
+			return
+		}
+
+	}
+}
+
+func callProcessorBox(processor Processor, appid string, boxname string, config Params, results chan<- Result) {
+	boxname = strings.ReplaceAll(boxname, "+", "%2B")
+	boxname = strings.ReplaceAll(boxname, "/", "%2F")
```

Comment:
This library is available, not sure if it's relevant to what you have here:
https://github.com/algorand/avm-abi/blob/main/apps/box.go

Reply:
that library makes key names for the KV store but does not prepare a B64 for URL encoding to the endpoints.

Reply:
replaced with net/url